### ISSUE TITLE
Creating environment from yaml fails when applying channel prefix 

### ISSUE THREAD

**Issue by FelixMoelder** [mamba-org/mamba#504]
mamba fails creating environments from yaml files when using a channel prefix e.g. `conda-forge::textacy` (see attachment for minimal example) returning the following error:

```
pkgs/main/noarch         [====================] (00m:00s) Done
pkgs/r/osx-64            [====================] (00m:00s) Done
pkgs/r/noarch            [====================] (00m:00s) Done
bioconda/osx-64          [====================] (00m:01s) Done
bioconda/noarch          [====================] (00m:01s) Done
pkgs/main/osx-64         [====================] (00m:03s) Done


Looking for: ['python=3.7', 'conda-forge::textacy']


ERROR   solver.cpp:66 Selected channel specific (or force-reinstall) job, but package is not available from channel. Solve job will fail.
```

This issue seems to be limited to creating environments from yaml-files, as `mamba create -n name conda-forge::textacy` works perfectly fine.
As this syntax works on command line and also when the channel has been defined in globally defined in conda config I would guess that the yaml-file is parsed correctly but the channel gets lost at some point and therefore is not checked out.

[environment-bazaarvoice-tagging.yml.zip](https://github.com/mamba-org/mamba/files/5233272/environment-bazaarvoice-tagging.yml.zip)

**Comment by alexeyegorov** [comment]
To add to the description above: creating the conda environment from this file with channel prefix works with conda command fine. ;)

**Comment by martinRenou** [comment]
This should have been fixed by #535
